File: Ejemplos/43-DockerAndTestContainers/src/main/java/dev/joseluisgs/repository/PersonasRepository.java
package dev.joseluisgs.repository;

import dev.joseluisgs.model.Persona;

import java.util.List;
import java.util.Optional;

public interface PersonasRepository {
    List<Persona> getAll();

    Optional<Persona> getById(long id);

    Persona create(Persona persona);

    Persona update(long id, Persona persona);

    boolean delete(long id);
}

File: Ejemplos/33-EjemplosTestRepositorioServicio/src/main/kotlin/personas/repository/PersonasRepository.kt
package dev.joseluisgs.personas.repository

import dev.joseluisgs.personas.model.Persona
import dev.joseluisgs.personas.repository.common.Repository
import java.util.*

interface PersonasRepository : Repository<UUID, Persona> {
    fun findByName(name: String): List<Persona>
}